src/main/java/ch/jalu/configme/resource/yaml/SnakeYamlNodeBuilderImpl.java

```diff
@@ -98,7 +98,23 @@ public void transferComments(@NotNull Node valueNode, @NotNull Node keyNode) {
     }
 
     protected @NotNull Node createStringNode(@NotNull String value) {
-        return new ScalarNode(Tag.STR, value, null, null, DumperOptions.ScalarStyle.PLAIN);
+        DumperOptions.ScalarStyle scalarStyle =
+            value.contains("\n")
+                ? DumperOptions.ScalarStyle.LITERAL // Used for strings that span multiple lines
+                : DumperOptions.ScalarStyle.PLAIN; // Used for single line string
+
+        if (scalarStyle == DumperOptions.ScalarStyle.LITERAL) {
+            value =
+                value
+                    .replace("\r", "") // removes carriage return
+                    .replaceAll(" +\\n", "\n"); // removes excess spaces before the \n
```

Comment:
I'm not sure this is a good idea to be doing, because I'd expect the properties to be exported with the values they were set to, i.e. the property value shouldn't be changed during the export.

Reply:
Happy new year! :smile:

Strings in YAML are a mess—I just saw that there are countless possibilities. I‘ll have to test it out. Does on of your tests cover the case of the string not being exported correctly without the transformation?

My thinking atm is that we can‘t assume which characters have meaning or not. It‘s a bit scary to strip things, especially _conditionally_. It doesn‘t help that carriage returns are removed, but only if the string contains `\n`. Not that I‘m suggesting to extend that condition lol

I saw from a quick glance at the spec that these formats add a new line at the end. This feature might take a while to make sure it checks all boxes 😅 I‘m not sure it‘s interesting to export values to a format that is not read identically. At the same time I definitely want something like this in ConfigMe, so we need to see what YAML allows us and what options we have to deal with that. ✈️

Reply:
If we keep carriage returns and skip the trimming, it seems that any values are exported _exactly_ as they are. What I mean by this is that strings with a new line at the end or the beginning is exported properly, even if the output is sometimes a bit unexpected. This is probably the best way to move forward with this, though I'll need to play around with reading files (Windows famously has CRLF lines) to see how much of an impact it is. 
I strongly believe ConfigMe shouldn't change a single byte in a property's value when it exports it; if this has unintended side effects, we need to introduce very specific options / property types. In a future version of ConfigMe we could even introduce options on the property somehow to allow to fine-tune the export for single properties, or e.g. by introducing property type extensions.